apps/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -415,17 +415,22 @@ export class ProjectBuildCache {
       }
     }
 
-    // Walk the tree in parallel
-    await Async.forEachAsync(
-      queue,
-      async ([relativePath, diskPath]: [string, string]) => {
-        const children: FolderItem[] = await FileSystem.readFolderItemsAsync(diskPath);
-        processChildren(relativePath, diskPath, children);
-      },
-      {
-        concurrency: 10
-      }
-    );
+    // Iterate through each folder level until we reach one that contains no subfolders.
+    while (nextLevelFolders.length > 0) {
+      const currentLevelFolders: [string, string][] = nextLevelFolders;
+      nextLevelFolders = [];
+      // Walk the tree in parallel
```

Comment:
This algorithm is somewhat complicated. Are you sure that it actually speeds anything up? It seems like it could have the potential to hammer the disk and maybe even bog the machine down (e.g. if multiple Rush processes are active).

1. We should do some basic measurements to confirm that this actually speeds things up significantly.
2. If it does, add a code comment summarizing what you measured, and explaining why we chose `concurrency: 10` as our parallelism. It doesn't need to be very formal, just a basic theory of operation for someone who might want to tune the constant.
3. Rush's default parallelism is based on the number of CPU cores. Maybe this should be too?

Reply:
It doesn't seem to make much of a difference. Changed it to be serial.